### PR TITLE
[DWS] Add support for cluster tag managment

### LINKED ISSUE #366: tags functionality in DWS SDK is not defined
Please add the API for GET /v1.0/{project_id}/clusters/tags as currently this is being used also by cloudmon and the collected metric is mis-interpreted as /cluster

### PR DESCRIPTION
**Pull request by Persimmonboy**
Closes #366

### CODE REVIEW

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by artem-lifshits — DISMISSED** [review]

**Review by vladimirvshivkov — DISMISSED** [review]

File: examples/dws/batch_create_cluster_tags.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# License for the specific language governing permissions and limitations
+# under the License.
+"""
+Batch create tags for a specified DWS cluster.
+"""
+import openstack
+
+openstack.enable_logging(True)
+conn = openstack.connect(cloud='otc')
+
```
Comment (Polina-Gubina) [inline examples/dws/batch_create_cluster_tags.py:22]: add please here:  sdk.register_otc_extensions(conn)
also you will need to add this import: from otcextensions import sdk

this comment is for all examples

File: examples/dws/create_cluster_tag.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+"""
+import openstack
+
+openstack.enable_logging(True)
+conn = openstack.connect(cloud='otc')
+
+cluster_id = 'cluster-uuid'
+tag = {'key': 'example_key', 'value': 'example_value'}
+
+result = conn.dws.create_cluster_tag(cluster_id, tag)
```
Comment (Polina-Gubina) [inline examples/dws/create_cluster_tag.py:26]: **tag

File: examples/dws/batch_delete_cluster_tags.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+openstack.enable_logging(True)
+conn = openstack.connect(cloud='otc')
+
+cluster_id = 'cluster-uuid'
+tags_to_delete = [
+    {'key': 'key1'},
+    {'key': 'key2'}
+]
+
+conn.dws.cluster_tags_batch_delete(cluster_id, tags_to_delete)
```
Comment (Polina-Gubina) [inline examples/dws/batch_delete_cluster_tags.py]: **tags_to_delete

File: examples/dws/batch_create_cluster_tags.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+cluster_id = 'cluster-uuid'
+tags_to_create = [
+    {'key': 'key1', 'value': 'value1'},
+    {'key': 'key2', 'value': 'value2'}
+]
+
+result = conn.dws.cluster_tags_batch_create(cluster_id, tags_to_create)
```
Comment (Polina-Gubina) [inline examples/dws/batch_create_cluster_tags.py]: **tags_to_create

File: otcextensions/sdk/dws/v1/_proxy.py [hunk 5]
```diff
@@ -341,3 +342,71 @@ def wait_for_cluster_scale_out(self, cluster, interval=5, wait=1800):
         if is_snapshotting:
             time.sleep(60)
             self.wait_for_cluster(cluster, interval, wait)
+
+    def list_cluster_tags(self, cluster):
```
Comment (Polina-Gubina) [inline otcextensions/sdk/dws/v1/_proxy.py]: rename please to cluster_tags

File: otcextensions/sdk/dws/v1/_proxy.py [hunk 6]
```diff
@@ -341,3 +342,71 @@ def wait_for_cluster_scale_out(self, cluster, interval=5, wait=1800):
...
+        return _tag.Tag().manage_tags_batch(self, cluster.id, tags, 'create')
+
+    def batch_delete_cluster_tags(self, cluster, tags):
+        """
+        Batch deletion of tags from a cluster.
+
+        :param cluster: Cluster ID or an instance of
+            `otcextensions.sdk.dws.v1.cluster.Cluster`.
+        :param tags: List of dictionaries of tags to delete.
+        :return: Response object.
```
Comment (Polina-Gubina) [inline otcextensions/sdk/dws/v1/_proxy.py]: should be None

Comment (Polina-Gubina) [reply otcextensions/sdk/dws/v1/_proxy.py]: or object Tag, not response

File: otcextensions/sdk/dws/v1/tag.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    value = resource.Body('value')
+
+    def manage_tags_batch(self, session, cluster_id, tags, action):
+        """
+        Manage tags in batch for a cluster (create or delete).
+        """
+        uri = utils.urljoin('clusters', cluster_id, 'tags/action')
+
+        body = {'action': action, 'tags': tags}
+        return session.post(uri, json=body)
```
Comment (Polina-Gubina) [inline otcextensions/sdk/dws/v1/tag.py]: it should return not 'clear' response, it must be object of class Tag

Comment (Polina-Gubina) [reply otcextensions/sdk/dws/v1/tag.py]: you need to translate response into object

File: otcextensions/tests/functional/sdk/dws/v1/test_tag.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    def tearDown(self):
+        """Clean up resources after testing."""
+        for tag_key in self.created_tags:
+            self.client.delete_cluster_tag(self.cluster_id, tag_key)
+        self.created_tags.clear()
+        super(TestClusterTags, self).tearDown()
+
+    def test_list_cluster_tags(self):
+        """Test listing all tags of a cluster."""
+        tags = list(self.client.list_cluster_tags(self.cluster_id))
```
Comment (Polina-Gubina) [inline otcextensions/tests/functional/sdk/dws/v1/test_tag.py]: you need to create instance and check that list returns more or equal 1, this test checks nothing

File: examples/dws/batch_create_cluster_tags.py (hunk 1 above)
Comment (Persimmonboy) [reply examples/dws/batch_create_cluster_tags.py:22]: Add the following to all examples:

<>
from otcextensions import sdk
<>
sdk.register_otc_extensions(conn)
<>

File: examples/dws/create_cluster_tag.py (hunk 2 above)
Comment (Persimmonboy) [reply examples/dws/create_cluster_tag.py:26]: Updated to **tag

File: examples/dws/batch_delete_cluster_tags.py (hunk 3 above)
Comment (Persimmonboy) [reply examples/dws/batch_delete_cluster_tags.py]: Updated to *tags_to_delete

File: examples/dws/batch_create_cluster_tags.py (hunk 4 above)
Comment (Persimmonboy) [reply examples/dws/batch_create_cluster_tags.py]: Updated to *tags_to_create

File: otcextensions/sdk/dws/v1/_proxy.py (hunk 5 above)
Comment (Persimmonboy) [reply otcextensions/sdk/dws/v1/_proxy.py]: Renamed, also in test, doc sections

File: otcextensions/sdk/dws/v1/_proxy.py (hunk 6 above)
Comment (Persimmonboy) [reply otcextensions/sdk/dws/v1/_proxy.py]: Changed response to 'otcextensions.sdk.dws.v1.tag.Tag'

File: otcextensions/tests/functional/sdk/dws/v1/test_tag.py (hunk 8 above)
Comment (Persimmonboy) [reply otcextensions/tests/functional/sdk/dws/v1/test_tag.py]: Updated the test

File: otcextensions/sdk/dws/v1/tag.py (hunk 7 above)
Comment (Persimmonboy) [reply otcextensions/sdk/dws/v1/tag.py]: Changed response into object

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by Polina-Gubina — APPROVED** [review]

**Pull request merged**